modules/devices/x86/processor.c

```diff
@@ -341,19 +341,23 @@ gchar *caches_summary(GSList * processors)
     return ret;
 }
 
-
+#define PROC_SCAN_READ_BUFFER_SIZE 896
 GSList *processor_scan(void)
 {
     GSList *procs = NULL, *l = NULL;
     Processor *processor = NULL;
     FILE *cpuinfo;
-    gchar buffer[512];
+    gchar buffer[PROC_SCAN_READ_BUFFER_SIZE];
 
     cpuinfo = fopen(PROC_CPUINFO, "r");
     if (!cpuinfo)
         return NULL;
 
-    while (fgets(buffer, 512, cpuinfo)) {
+    while (fgets(buffer, PROC_SCAN_READ_BUFFER_SIZE, cpuinfo)) {
```

Comment:
Might be better to use g_file_read_contents() and split on '\n'.  It'll allocate more memory and fragment the heap more, but then we'll not hit these limits.